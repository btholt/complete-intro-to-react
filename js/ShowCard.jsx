import React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: 32%;
  border: 2px solid #333;
  border-radius: 4px;
  margin-bottom: 25px;
  padding-right: 10px;
  overflow: hidden;
`;

const Image = styled.img`
  width: 46%;
  float: left;
  margin-right: 10px;
`;

const ShowCard = props => (
  <Wrapper>
    <Image alt={`${props.title} Show Poster`} src={`/public/img/posters/${props.poster}`} />
    <div>
      <h3>{props.title}</h3>
      <h4>({props.year})</h4>
      <p>{props.description}</p>
    </div>
  </Wrapper>
);

ShowCard.propTypes = {
  poster: string.isRequired,
  title: string.isRequired,
  year: string.isRequired,
  description: string.isRequired,
};

export default ShowCard;
