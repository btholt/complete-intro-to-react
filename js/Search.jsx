// @flow

import React, { Component } from 'react';
import ShowCard from './ShowCard';

class Search extends Component {
  state = {
    searchTerm: ''
  };

  props: {
    shows: Array<Show>
  };

  handleSearchTermChange = (event: SyntheticKeyboardEvent & { target: HTMLInputElement }) => {
    this.setState({ searchTerm: event.target.value });
  };

  render() {
    return (
      <div className="search">
        <header>
          <h1>svideo</h1>
          <input
            onChange={this.handleSearchTermChange}
            value={this.state.searchTerm}
            type="text"
            placeholder="Search"
          />
        </header>
        <div>
          {this.props.shows
            .filter(
              show =>
                `${show.title} ${show.description}`.toUpperCase().indexOf(this.state.searchTerm.toUpperCase()) >= 0
            )
            .map(show => <ShowCard key={show.imdbID} {...show} />)}
        </div>
      </div>
    );
  }
}

export default Search;
