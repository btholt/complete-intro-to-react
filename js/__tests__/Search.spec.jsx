import React from 'react';
import { shallow } from 'enzyme';
import preload from '../../data.json';
import Search from '../Search';
import ShowCard from '../ShowCard';

test('Search renders correctly', () => {
  const component = shallow(<Search shows={preload.shows} />);
  expect(component).toMatchSnapshot();
});

test('Search should render correct amount of shows', () => {
  const component = shallow(<Search shows={preload.shows} />);
  expect(component.find(ShowCard).length).toEqual(preload.shows.length);
});

test('Search should render correct amount of shows based on search term', () => {
  const searchWord = 'black';
  const component = shallow(<Search shows={preload.shows} />);
  component.find('input').simulate('change', { target: { value: searchWord } });
  const showCount = preload.shows.filter(
    show => `${show.title} ${show.description}`.toUpperCase().indexOf(searchWord.toUpperCase()) >= 0
  ).length;
  expect(component.find(ShowCard).length).toEqual(showCount);
});
