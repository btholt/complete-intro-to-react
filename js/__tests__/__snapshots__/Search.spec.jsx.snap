// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <header>
    <h1>
      svideo
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </header>
  <div>
    <ShowCard
      show={
        Object {
          "description": "Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; \\"Earnest 'Earn' Marks,\\" an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.",
          "imdbID": "tt4288182",
          "poster": "a.jpg",
          "title": "Atlanta",
          "trailer": "MpEdJ-mmTlY",
          "year": "2008–2013",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "U.S. Attorney Chuck Rhoades goes after hedge fund king, Bobby \\"Axe\\" Axelrod in a battle between two powerful New York figures.",
          "imdbID": "tt4270492",
          "poster": "b.jpg",
          "title": "Billions",
          "trailer": "_raEUMLL-ZI",
          "year": "2016–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A television anthology series that shows the dark side of life and technology.",
          "imdbID": "tt2085059",
          "poster": "bm.jpg",
          "title": "Black Mirror",
          "trailer": "jDiYGjp5iFg",
          "year": "2011–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A high school chemistry teacher diagnosed with inoperable lung cancer turns to manufacturing and selling methamphetamine in order to secure his family's future.",
          "imdbID": "tt0903747",
          "poster": "bb.jpg",
          "title": "Breaking Bad",
          "trailer": "XZ8daibM3AE",
          "year": "2008–2013",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years.",
          "imdbID": "tt0944947",
          "poster": "got.jpg",
          "title": "Game of Thrones",
          "trailer": "giYeaKsXnsI",
          "year": "2011–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A bipolar CIA operative becomes convinced a prisoner of war has been turned by al-Qaeda and is planning to carry out a terrorist attack on American soil.",
          "imdbID": "tt1796960",
          "poster": "h.jpg",
          "title": "Homeland",
          "trailer": "KyFmS3wRPCQ",
          "year": "2011–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A Congressman works with his equally conniving wife to exact revenge on the people who betrayed him.",
          "imdbID": "tt1856010",
          "poster": "hoc.jpg",
          "title": "House of Cards",
          "trailer": "NTzycsqxYJ0",
          "year": "2013–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "The personal and professional life of Dev, a 30-year-old actor in New York.",
          "imdbID": "tt4635276",
          "poster": "mon.jpg",
          "title": "Master of None",
          "trailer": "6bFvb3WKISk",
          "year": "2015–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "A chronicled look at the criminal exploits of Colombian drug lord Pablo Escobar.",
          "imdbID": "tt2707408",
          "poster": "n.jpg",
          "title": "Narcos",
          "trailer": "U7elNhHwgBU",
          "year": "2015–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "The story of Piper Chapman, a woman in her thirties who is sentenced to fifteen months in prison after being convicted of a decade-old crime of transporting money for her drug-dealing girlfriend.",
          "imdbID": "tt2372162",
          "poster": "oitnb.jpg",
          "title": "Orange Is the New Black",
          "trailer": "th8WT_pxGqg",
          "year": "2013–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper.",
          "imdbID": "tt2575988",
          "poster": "sv.jpg",
          "title": "Silicon Valley",
          "trailer": "69V__a49xtw",
          "year": "2014–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "When a young boy disappears, his mother, a police chief, and his friends must confront terrifying forces in order to get him back.",
          "imdbID": "tt4574334",
          "poster": "st.jpg",
          "title": "Stranger Things",
          "trailer": "9Egf5U8xLo8",
          "year": "2016–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Two Soviet intelligence agents pose as a married couple to spy on the American government.",
          "imdbID": "tt2149175",
          "poster": "ta.jpg",
          "title": "The Americans",
          "trailer": "HjuUkbhsI24",
          "year": "2013–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Former Senator Selina Meyer finds that being Vice President of the United States is nothing like she hoped and everything that everyone ever warned her about.",
          "imdbID": "tt1759761",
          "poster": "v.jpg",
          "title": "Veep",
          "trailer": "I2HoCG_iyhU",
          "year": "2012–",
        }
      }
    />
    <ShowCard
      show={
        Object {
          "description": "Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence.",
          "imdbID": "tt0475784",
          "poster": "ww.jpg",
          "title": "Westworld",
          "trailer": "eX3u0IlBBO4",
          "year": "2016–",
        }
      }
    />
  </div>
</div>
`;
